src/Microsoft.DotNet.VersionTools/Automation/GitHubApi/GitHubClient.cs

```diff
@@ -0,0 +1,211 @@
+            {
+                title = title,
+                body = description,
+                head = $"{headBranch.Project.Owner}:{headBranch.Name}",
+                @base = baseBranch.Name
+            }, Formatting.Indented);
+
+            string pullUrl = $"https://api.github.com/repos/{baseBranch.Project.Segments}/pulls";
+
+            var bodyContent = new StringContent(createPrBody);
+            using (HttpResponseMessage response = await _httpClient.PostAsync(pullUrl, bodyContent))
+            {
+                response.EnsureSuccessStatusCode();
+
+                Trace.TraceInformation($"Created pull request.");
+                Trace.TraceInformation($"Pull request page: {await GetPullRequestUrlAsync(response)}");
+            }
+        }
+
+        public async Task UpdateGitHubPullRequestAsync(
+            GitHubProject project,
+            int number,
+            string title = null,
+            string body = null,
+            string state = null)
+        {
+            var updatePrBody = new JObject();
+
+            if (title != null)
+            {
+               updatePrBody.Add(new JProperty("title", title));
+            }
+            if (body != null)
+            {
+                updatePrBody.Add(new JProperty("body", body));
+            }
+            if (state != null)
+            {
+                updatePrBody.Add(new JProperty("state", state));
+            }
+
+            string url = $"https://api.github.com/repos/{project.Segments}/pulls/{number}";
+
+            HttpRequestMessage request = new HttpRequestMessage(new HttpMethod("PATCH"), url);
+            request.Content = new StringContent(JsonConvert.SerializeObject(updatePrBody));
+
+            using (HttpResponseMessage response = await _httpClient.SendAsync(request))
+            {
+                response.EnsureSuccessStatusCode();
+
+                Trace.TraceInformation($"Updated pull request #{number}.");
+                Trace.TraceInformation($"Pull request page: {await GetPullRequestUrlAsync(response)}");
+            }
+        }
+
+        public async Task<GitHubPullRequest> FirstPullRequestWithHeadOrDefaultAsync(
+            GitHubProject project,
+            string headPrefix,
+            string author,
+            string sortType = "created")
+        {
+            int pullRequestNumber;
+
+            // First: find the number of the pull request.
+            string queryString = $"repo:{project.Segments}+head:{headPrefix}+author:{author}+state:open";
+            string queryUrl = $"https://api.github.com/search/issues?q={queryString}&sort={sortType}&order=desc";
+
+            using (HttpResponseMessage response = await _httpClient.GetAsync(queryUrl))
+            {
+                response.EnsureSuccessStatusCode();
+
+                JObject responseContent = JObject.Parse(await response.Content.ReadAsStringAsync());
+
+                int totalCount = responseContent["total_count"].Value<int>();
```

Comment:
Do you think it would make sense to have an object model for this object?  Like we do for GitHubPullRequest.